File: src/models/config/prompt_dictionary_interface.py
import json
import os


class PromptDictionary:
  def __init__(self):
    with open('prompt_dictionary.json', encoding='utf-8') as f:
      self.prompt_dictionary = json.load(f)

  def get_prompt(self, path_or_version: str) -> str:

    if os.path.isfile(path_or_version):
      with open(path_or_version, 'r', encoding='utf-8') as file:
        return file.read()
    else:
      prompt_path = self._get_value_from_path(
          self.prompt_dictionary, path_or_version)
      if os.path.isfile(prompt_path):
        with open(prompt_path, 'r', encoding='utf-8') as file:
          return file.read()
      else:
        return "File not found"

  def _get_value_from_path(self, d, path_str):
    temp = d
    for key in path_str.split('.'):
      if key in temp:
        temp = temp[key]
      else:
        return "Key not found"
    return temp

File: src/models/config/prompt_dictionary_interface.ts
import fs from 'fs';

type PromptDictionaryType = {
  [key: string]: string | PromptDictionaryType;
};

export class PromptDictionary {
  private promptDictionary: PromptDictionaryType;

  constructor() {
    const data = fs.readFileSync('prompt_dictionary.json', 'utf-8');
    this.promptDictionary = JSON.parse(data) as PromptDictionaryType;
  }

  public getPrompt(pathOrVersion: string): string {
    if (fs.existsSync(pathOrVersion)) {
      return fs.readFileSync(pathOrVersion, 'utf-8');
    } else {
      const promptPath = this.getValueFromPath(this.promptDictionary, pathOrVersion);
      if (typeof promptPath === 'string' && fs.existsSync(promptPath)) {
        return fs.readFileSync(promptPath, 'utf-8');
      } else {
        return 'File not found';
      }
    }
  }

  private getValueFromPath(obj: PromptDictionaryType, pathStr: string): string | PromptDictionaryType {
    let temp: string | PromptDictionaryType = obj;
    for (const key of pathStr.split('.')) {
      if (typeof temp === 'object' && key in temp) {
        temp = temp[key] as string | PromptDictionaryType;
      } else {
        return 'Key not found';
      }
    }
    return temp;
  }
}
